aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+        .build());
+    String dbLocationUri = response.database().locationUri();
+    if (dbLocationUri != null) {
+      return String.format("%s/%s", dbLocationUri, tableIdentifier.name());
+    }
+
+    return String.format(
+        "%s/%s.db/%s",
+        warehousePath,
+        IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()),
+        tableIdentifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    namespaceExists(namespace);
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<TableIdentifier> results = new ArrayList<>();
+    while (true) {
+      GetTablesResponse response = glue.getTables(GetTablesRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.toDatabaseName(namespace))
+          .nextToken(nextToken)
+          .build());
+      if (response.hasTableList()) {
+        results.addAll(response.tableList().stream()
+            .map(GlueToIcebergConverter::toTableId)
+            .collect(Collectors.toList()));
+        nextToken = response.nextToken();
+      }
+      if (nextToken == null) {
+        break;
+      }
+    }
+    return results;
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    try {
+      dropTable(newTableOps(identifier), identifier, purge);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  private TableMetadata dropTable(TableOperations ops, TableIdentifier identifier, boolean purge) {
+    TableMetadata lastMetadata = null;
+    if (ops.current() != null) {
+      lastMetadata = ops.current();
+    }
+
+    glue.deleteTable(DeleteTableRequest.builder()
+        .catalogId(catalogId)
+        .databaseName(IcebergToGlueConverter.toDatabaseName(identifier.namespace()))
+        .name(identifier.name())
+        .build());
+
+    if (purge && lastMetadata != null) {
+      CatalogUtil.dropTableData(ops.io(), lastMetadata);
+    }
+
+    LOG.info("Dropped table: {}", identifier);
+    return lastMetadata;
+  }
+
+  /**
+   * Rename table in Glue is a drop table and create table.
+   * @param from identifier of the table to rename
+   * @param to new table name
+   */
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    TableOperations ops = newTableOps(from);
+    // keep metadata
+    TableMetadata tableMetadata = dropTable(ops, from, false);
+    // use the same info to create the new table
+    createTable(to, tableMetadata.schema(), tableMetadata.spec(), tableMetadata.properties());
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    glue.createDatabase(CreateDatabaseRequest.builder()
+        .catalogId(catalogId)
+        .databaseInput(IcebergToGlueConverter.toDatabaseInput(namespace, metadata))
+        .build());
```

Comment:
Seems like this operation can throw as well. Since the `BaseMetastoreCatalog` doesn't declare thi with a checked exception, you might want to document what it can throw that you know of or wrap in `try...catch` (though arguably that could be up the calling code).

Reply:
Based on my observation of other catalogs, we are not handling runtime exceptions, and aws clients only throw runtime exception. That is why I do not try catch anything here. I can probably add a check for AlreadyExistsException.